test/client/api.js

```diff
@@ -61,11 +61,20 @@ ClientApi.prototype.doRequest = function (method, url, token, payload, headers)
 
     this.emit('endRequest', options, err, res)
     if (err || body.error || res.statusCode !== 200) {
-      d.reject(err || body)
+      return d.reject(err || body)
     }
-    else {
-      d.resolve(body)
+
+    var allowedOrigin = res.headers['access-control-allow-origin']
+    if (allowedOrigin) {
+      // Requiring config outside this condition causes the local tests to fail
+      // because tokenLifetimes.passwordChangeToken is -1
+      var config = require('../../config')
```

Comment:
Not that I can think of; subsequent calls just return the cached result from the first invocation. It's no different to what happens when lots of different modules require it separately.